### PR TITLE
feat/create component toast

### PR DESCRIPTION
**Pull request by SylviaGonschior**
added unit tests

### CODE REVIEW

File: packages/components/src/components/toast/toast.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      this.theme && `toast--theme-${this.theme}`,
+      this.variant && `toast--variant-${this.variant}`
+    );
+  }
+
+  private autohideTime = 5000;
+
+  onCloseToast = () => {
+    this.opened = false;
+    console.log("onCloseToast");
```
Comment (eeegor) [inline packages/components/src/components/toast/toast.tsx]: Please remove `console.log` before merge

File: packages/components/src/components/toast/toast.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  @Method()
+  async openToast() {
+    this.opened = true;
+  }
+
+  getTime = () => {
+    const formattedTime =
+      this.time &&
+      formatDistance(subSeconds(this.time, 3), new Date(), { addSuffix: true });
+    console.log(formattedTime);
```
Comment (eeegor) [inline packages/components/src/components/toast/toast.tsx]: Please remove `console.log` before merge

File: packages/components/src/components/toast/toast.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    return formattedTime;
+  };
+
+  onHideToast = () => {
+    if (this.autohide !== false) {
```
Comment (eeegor) [inline packages/components/src/components/toast/toast.tsx]: Currently this event fires even if the `toast` is not opened

File: packages/components/src/components/toast/toast.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      setTimeout(this.onCloseToast, this.autohideTime);
```
Comment (eeegor) [inline packages/components/src/components/toast/toast.tsx]: You need to make sure that the `toast` only can be opened once, and that the timeout is also only fired once. Currently if you call `openToast()` multiple times it creates multiple timeouts which leads to buggy behaviour. Also you'll need to `clearTimeout()` once it's finished or aborted (component not on page anymore e.g. page switch) see https://www.w3schools.com/jsref/met_win_cleartimeout.asp for more details. 

You can debug the timeout doing something similar to this: 
```javascript
if (this.opened && this.autohide !== false) {
    const myTimeout = setTimeout(this.onCloseToast, this.autohideTime);
    console.log('myTimeout', myTimeout)
    return myTimeout;
} else {
    return null;
}
```

File: test/__mocks__/date-fns.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const formatDistance = (date, baseDate) => `${+date} [${baseDate}]`;
+
+export const subSeconds = (date, amount) => `${+date} [${amount}]`;
```
Comment (eeegor) [inline test/__mocks__/date-fns.js]: `date-fns` should be in `devDependencies`, it's currently missing 😉 

You can add it with `yarn add date-fns --dev`

Comment (SylviaGonschior) [reply test/__mocks__/date-fns.js]: oh, I did it several times

Comment (eeegor) [reply test/__mocks__/date-fns.js]: image

still missing...

Comment (eeegor) [reply test/__mocks__/date-fns.js]: Maybe you forgot to commit ;)

File: packages/components/src/components/toast/toast.tsx (hunk 4 above)
Comment (eeegor) [reply packages/components/src/components/toast/toast.tsx]: image
the timeout is still firing multiple times..

**Pull request merged**